SS2-Project/Assets/Starstorm2/Modules/Characters/Survivors/Knight.cs

```diff
@@ -88,6 +116,70 @@ public void ModifyStatArguments(RecalculateStatsAPI.StatHookEventArgs args)
                 args.moveSpeedReductionMultAdd += 0.6f;
             }
         }
+
+
+        public class KnightSpecialEmpowerBuff : BaseBuffBehaviour, IBodyStatArgModifier
+        {
+            [BuffDefAssociation]
+            private static BuffDef GetBuffDef() => SS2Content.Buffs.bdKnightSpecialPowerBuff;
+            public void ModifyStatArguments(RecalculateStatsAPI.StatHookEventArgs args)
+            {
+                args.baseJumpPowerAdd += 0.7f;
+                args.baseMoveSpeedAdd += 0.4f;
+            }
+        }
+
+        public class KnightSpecialSlowEnemiesBuff : BaseBuffBehaviour, IBodyStatArgModifier
+        {
+            [BuffDefAssociation]
+            private static BuffDef GetBuffDef() => SS2Content.Buffs.bdKnightSpecialSlowBuff;
+            public void ModifyStatArguments(RecalculateStatsAPI.StatHookEventArgs args)
+            {
+                args.attackSpeedReductionMultAdd += 2;
+                args.moveSpeedReductionMultAdd += 2;
+            }
+        }
+
+        public class KnightParryBuff : BaseBuffBehaviour, IOnIncomingDamageServerReceiver
+        {
+            [BuffDefAssociation]
+            private static BuffDef GetBuffDef() => SS2Content.Buffs.bdParry;
+
+            // Called when the body with this buff takes damage
+            public void OnIncomingDamageServer(DamageInfo damageInfo)
+            {
+                if (damageInfo.attacker != CharacterBody)
```

Comment:
Check if the behaviour is enabled by returning early after checking if "HasAnyStacks" is true.

Excerpt from the documentation on buff behaviour:

``
Due to the nature of Buffs being Ephemeral, this Behaviour is not destroyed when the buff is removed, but instead it becomes disabled. Due to this reason, most, if not all behaviour interfaces will still get called even if the behaviour itself is disabled. For this reason it is recommended to break from the function early by checking if <see cref="HasAnyStacks"/> is false``